### PR TITLE
Fix live streaming and update for 2017 replays

### PR DESCRIPTION
**Pull request by glennguy**

### CODE REVIEW

File: resources/lib/telstra_auth.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import json
+import urlparse
+import config
+import comm
+import utils
+from exception import AFLVideoException
+
+class SortedHTTPAdapter(requests.adapters.HTTPAdapter):
+    def add_headers(self, request, **kwargs):
+        if 'header_order' in globals():
```
Comment (andybotting) [inline resources/lib/telstra_auth.py]: Can we remove the globals? Could we just have a header_order variable at the top of this file maybe?

File: resources/lib/telstra_auth.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    session = requests.Session()
+    session.verify = False
+    session.mount("https://", SortedHTTPAdapter())
+    global header_order
+        
+    # Send our first login request to AFL API, recieve (unactivated) token
+    # and 'msisdn' URL
+
+    session.headers = {'x-media-mis-token': comm.fetch_token()}
+    auth_resp = session.post('http://api.sub.afl.com.au/cfs-premium/users?paymentMethod=ONE_PLACE')
```
Comment (andybotting) [inline resources/lib/telstra_auth.py]: Should this URL be in config.py?

File: resources/settings.xml [hunk 3]
```diff
@@ -6,7 +6,9 @@
     </category>
     <category label="20002">
 		<setting id="LIVE_QUALITY" type="enum" label="10006" values="Lowest (300k/s)|Low (450k/s)|Medium (660k/s)|High (1Mb/s)|Highest (1.6Mb/s)" default="4" />
+        <setting id="SUBSCRIPTION_TYPE" type="enum" label="10007" values="Paid|Free offer" default="1" />
```
Comment (andybotting) [inline resources/settings.xml]: Inconsistent spacing

File: resources/lib/telstra_auth.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    saml_base64 = signon.text[start_index:end_index].replace(
+                                                    '&#xa;', '%0D%0A').replace(
+                                                    '&#x3d;', '%3D').replace(
+                                                    '&#x2b;', '%2B')
+    
+    # Send the SAML login data and retrieve the auth token from the response
+    session.headers = config.SAML_LOGIN_HEADERS
+    session.cookies.set('saml_request_path', msisdn_url)
+    saml_login = session.post(config.SAML_LOGIN_URL, 
+                            data='SAMLResponse={0}'.format(saml_base64))
```
Comment (andybotting) [inline resources/lib/telstra_auth.py]: doesn't line up

File: resources/lib/telstra_auth.py [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    if 'errorcode' in signon_query:
+        if signon_query['errorcode'] == '0':
+             raise AFLVideoException('Please enter your username in the settings')
+        if signon_query['errorcode'] == '1':
+             raise AFLVideoException('Please enter your password in the settings')
+        if signon_query['errorcode'] == '2':
+             raise AFLVideoException('Please enter your username and password in the settings')
+        if signon_query['errorcode'] == '3':
+             raise AFLVideoException('Please check your username and password in the settings')
+    start_index = signon.text.find('SAMLResponse')+21
```
Comment (andybotting) [inline resources/lib/telstra_auth.py]: magic numbers are scary. is there a saml library we should use for this, or not worth it?

File: resources/settings.xml (hunk 3 above)
Comment (glennguy) [inline resources/settings.xml]: Thanks notepad++ 😡

File: resources/lib/telstra_auth.py (hunk 5 above)
Comment (glennguy) [inline resources/lib/telstra_auth.py]: This is just grabbing a value from an HTML tag - eg ``` <input type="HIDDEN" name="SAMLResponse" value="PHNhbW.......``` 
The other option I was thinking of was a custom HTML parser class to just grab that tag. What do you reckon?

File: resources/lib/telstra_auth.py (hunk 2 above)
Comment (glennguy) [inline resources/lib/telstra_auth.py]: yep

File: resources/lib/telstra_auth.py (hunk 1 above)
Comment (glennguy) [inline resources/lib/telstra_auth.py]: Just remembered that only the NRL auth process needed the specific header orders and even now it seems they've relaxed that strange requirement so I'll just remove altogether.

File: resources/lib/telstra_auth.py (hunk 5 above)
Comment (andybotting) [inline resources/lib/telstra_auth.py]: Could use BeautifulSoup if you're parsing HTML. It's very good

**Comment by andybotting** [comment]
`                                              File "/home/andy/.kodi/addons/plugin.video.afl-video/resources/lib/play.py", line 39, in play
                                                login_token = ooyalahelper.get_afl_user_token()
                                              File "/home/andy/.kodi/addons/plugin.video.afl-video/resources/lib/ooyalahelper.py", line 134, in get_afl_user_token
                                                return telstra_auth.get_token(username, password, phone_number)
                                              File "/home/andy/.kodi/addons/plugin.video.afl-video/resources/lib/telstra_auth.py", line 21, in get_token
                                                jsondata = json.loads(auth_resp.text)
                                            NameError: global name 'auth_resp' is not defined
`

**Comment by glennguy** [comment]
Forgot to tick the box for the file in github desktop after I made the last change... 😩

**Pull request merged**